modules/nextflow/src/main/groovy/nextflow/config/ConfigBuilder.groovy

```diff
@@ -735,5 +735,21 @@ class ConfigBuilder {
         buildConfigObject().toMap()
     }
 
+    /**
+     * Merge two maps recursively avoiding keys to be overwritten
+     *
+     * @param lmap
+     * @param rmap
+     * @return a map resulting of merging lmap and rmap
+     */
+    def mergeMaps(Map lmap, Map rmap) {
```

Comment:
Let's declare this `protected Map mergeMaps(Map lmap, Map rmap)` ie add explicit return type